aiida/cmdline/commands/cmd_profile.py

```diff
@@ -133,69 +135,37 @@ def profile_setdefault(profile):
-    'include_database',
-    default=True,
-    show_default=True,
-    help='Include deletion of associated database.'
-)
-@click.option(
-    '--include-db-user/--skip-db-user',
-    'include_database_user',
-    default=False,
-    show_default=True,
-    help='Include deletion of associated database user.'
-)
+@options.FORCE(help='Skip any prompts for confirmation.')
 @click.option(
-    '--include-repository/--skip-repository',
-    default=True,
-    show_default=True,
-    help='Include deletion of associated file repository.'
+    '--delete-data/--keep-data',
+    default=None,
+    help='Whether to delete the storage with all its data or not. This flag has to be explicitly specified'
 )
 @arguments.PROFILES(required=True)
-def profile_delete(force, include_config, include_database, include_database_user, include_repository, profiles):
+def profile_delete(force, delete_data, profiles):
     """Delete one or more profiles.
 
     The PROFILES argument takes one or multiple profile names that will be deleted. Deletion here means that the profile
-    will be removed including its file repository and database. The various options can be used to control which parts
-    of the profile are deleted.
+    will be removed from the config file. If ``--delete-storage`` is specified, the storage containing all data is also
+    deleted.
     """
-    if not include_config:
-        echo.echo_deprecated('the `--skip-config` option is deprecated and is no longer respected.')
-
-    for profile in profiles:
-
-        includes = {
-            'database': include_database,
-            'database user': include_database_user,
-            'file repository': include_repository
-        }
+    if force and delete_data is None:
+        raise click.BadParameter(
+            'When the `-f/--force` flag is used either `--delete-data` or `--keep-data` has to be explicitly specified.'
+        )
 
-        if not all(includes.values()):
-            excludes = [label for label, value in includes.items() if not value]
-            message_suffix = f' excluding: {", ".join(excludes)}.'
-        else:
-            message_suffix = '.'
+    if not force and delete_data is None:
+        echo.echo_warning('Do you also want to permanently delete all data?', nl=False)
+        delete_data = click.confirm('', default=False)
 
-        echo.echo_warning(f'deleting profile `{profile.name}`{message_suffix}')
-        echo.echo_warning('this operation cannot be undone, ', nl=False)
+    for profile in profiles:
+        suffix = click.style('including' if delete_data else 'without', fg='red', bold=True)
+        echo.echo_warning(f'Deleting profile `{profile.name}`, {suffix} all data.')
 
-        if not force and not click.confirm('are you sure you want to continue?'):
-            echo.echo_report(f'deleting of `{profile.name} cancelled.')
+        echo.echo_warning('This operation cannot be undone, are you sure you want to continue?', nl=False)
```

Comment:
This prompt should only appear when the force flag is not used, right?

```
❯ verdi profile delete tmp -f --delete-data
Warning: Deleting profile `tmp`, including all data.
Warning: This operation cannot be undone, are you sure you want to continue?Report: Deleted storage directory at `/Users/mbercx/project/core/.aiida/repository/sqlite_dos_5b7189522ae34118b151f65f618d150c`.
Report: Data storage deleted, configuration was: {'filepath': '/Users/mbercx/project/core/.aiida/repository/sqlite_dos_5b7189522ae34118b151f65f618d150c'}
Warning: `tmp` was the default profile, no profiles remain to set as default.
Success: Profile `tmp` was deleted.
```

Reply:
Damn, my desire to have prompts prefixed with the nice `Warning:` strikes again...

Reply:
Fixed. By the way, I changed the flags to `--delete-data/--keep-data` because I think that is going to be more clear for most users, rather than "storage" which is kind of vague. You agree?

Reply:
Yeah, I like it better! 👍